### PR TITLE
Feat: Archive api calls

### PR DESCRIPTION
**Pull request by Rohitprasad83**
## Added archive API calls
- get all archive notes
- add to the archive
- restore a note from the archive
- delete an archive note

## Will Add later:-
- will add the note handlers to different files
- Cleanup overall file structure
- use a different icon for restoring a note.

## Gif

chrome_ayNdeZYOMU

### CODE REVIEW

File: src/pages/archive/Archive.jsx [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
+import { useNote, useAuth } from 'context/index'
+import axios from 'axios'
+import { Note } from 'components/main/Note'
+
 function Archive() {
+  const { encodedToken } = useAuth()
+
+  const { archiveNotes, setArchiveNotes } = useNote()
+  useEffect(() => {
+    ;(async () => {
```
Comment (Naveenchandar) [inline src/pages/archive/Archive.jsx:13]: Please remove semicolon

**Review by Naveenchandar — APPROVED** [review]
Good usage of authentication context
Folder structure neatly maintained
Short and crisp components.

**Review by Aditi0505 — APPROVED** [review]
- Clean and readable code.
- Good use of toast component for error handling
- Good PR description

File: src/pages/archive/Archive.jsx (hunk 1 above)
Comment (Rohitprasad83) [reply src/pages/archive/Archive.jsx:13]: Prettier is putting it, i tried to remove it but was not able to, probably some problem in prettier setting

**Pull request merged**